dbos/workflows_test.go

```diff
@@ -1961,3 +1962,376 @@ func TestSleep(t *testing.T) {
+		}
+
+		// Wait for the parent workflow to complete and get the result
+		result, err := handle.GetResult()
+		if !errors.Is(err, context.DeadlineExceeded) {
+			t.Fatalf("Expected deadline exceeded error, got: %v", err)
+		}
+		if result != "" {
+			t.Fatalf("expected result to be an empty string, got '%s'", result)
+		}
+
+		// Check the workflow status: should be cancelled
+		status, err := handle.GetStatus()
+		if err != nil {
+			t.Fatalf("failed to get workflow status: %v", err)
+		}
+		if status.Status != WorkflowStatusCancelled {
+			t.Fatalf("expected workflow status to be WorkflowStatusCancelled, got %v", status.Status)
+		}
+	})
+
+	detachedChild := func(ctx DBOSContext, timeout time.Duration) (string, error) {
+		select {
+		case <-ctx.Done():
+			return "", ctx.Err()
+		case <-time.After(timeout):
+		}
+		return "detached-step-completed", nil
+	}
+	RegisterWorkflow(dbosCtx, detachedChild)
+
+	detachedChildWorkflowParent := func(ctx DBOSContext, timeout time.Duration) (string, error) {
+		childCtx := WithoutCancel(ctx)
+		childHandle, err := RunAsWorkflow(childCtx, detachedChild, timeout*2)
+		if err != nil {
+			t.Fatalf("failed to start child workflow: %v", err)
+		}
+
+		// Wait for the child workflow to complete
+		result, err := childHandle.GetResult()
+		if err != nil {
+			t.Fatalf("failed to get result from child workflow: %v", err)
+		}
+		// Check the child workflow status: should be cancelled
+		status, err := childHandle.GetStatus()
+		if err != nil {
+			t.Fatalf("failed to get child workflow status: %v", err)
+		}
+		if status.Status != WorkflowStatusSuccess {
+			t.Fatalf("expected child workflow status to be WorkflowStatusSuccess, got %v", status.Status)
+		}
+		// The child spun for timeout*2 so ctx.Err() should be context.DeadlineExceeded
+		return result, ctx.Err()
+	}
+	RegisterWorkflow(dbosCtx, detachedChildWorkflowParent)
+
+	t.Run("ChildWorkflowDetached", func(t *testing.T) {
+		timeout := 500 * time.Millisecond
+		cancelCtx, cancelFunc := WithTimeout(dbosCtx, timeout)
+		defer cancelFunc()
+		handle, err := RunAsWorkflow(cancelCtx, detachedChildWorkflowParent, timeout)
+		if err != nil {
+			t.Fatalf("failed to start parent workflow with detached child: %v", err)
+		}
+
+		// Wait for the parent workflow to complete and get the result
+		result, err := handle.GetResult()
+		if !errors.Is(err, context.DeadlineExceeded) {
+			t.Fatalf("Expected deadline exceeded error, got: %v", err)
+		}
+		if result != "detached-step-completed" {
+			t.Fatalf("expected result to be 'detached-step-completed', got '%s'", result)
+		}
+
+		// Check the workflow status: should be cancelled
+		status, err := handle.GetStatus()
+		if err != nil {
+			t.Fatalf("failed to get workflow status: %v", err)
+		}
+		if status.Status != WorkflowStatusCancelled {
+			t.Fatalf("expected workflow status to be WorkflowStatusCancelled, got %v", status.Status)
+		}
+	})
+
+	t.Run("RecoverWaitForCancelWorkflow", func(t *testing.T) {
```

Comment:
I don't find this test super useful -- yes it checks that you can recover (run again) a workflow that has a timeout, and that this new invocation also times out, but it's exactly what you'd expect.